### PR TITLE
chore: remove constraints related to db integrity errors

### PR DESCRIPTION
**Pull request by schenedx**
## Description
Remove the constraints on `moto` and `idna` because these can be address by unit test code changes.

### CODE REVIEW

File: requirements/common_constraints.txt [hunk 1]
```diff
@@ -13,6 +13,16 @@
...
+# This is a temporary solution to override the real common_constraints.txt
+# In edx-lint, until the pyjwt constraint in edx-lint has been removed.
+# See BOM-2721 for more details.
+# Below is the copied and edited version of common_constraints
+
+# This is a temporary solution to override the real common_constraints.txt
+# In edx-lint, until the pyjwt constraint in edx-lint has been removed.
+# See BOM-2721 for more details.
+# Below is the copied and edited version of common_constraints
+
```
Comment (matthugs) [inline requirements/common_constraints.txt]: it looks like this comment is getting duplicatively added every time we run, e.g. `make upgrade`? We should maybe report that to BOM so they can help us avoid needing to think about it, and remove this from our diffs.

**Review by matthugs — CHANGES_REQUESTED** [review]
Do we need `moto` outside of the test requirements? I wonder if we can just eliminate it from a number of requirements files (except for test) while we're looking at this?

**Comment by schenedx** [comment]
@matthugs the `moto` library is only specified in the `test.in` file. However, `local.in`, `devstack.in` and `monitoring/requirements.in` all import `test.in`. That's why moto is installed on those. I don't want to expand the scope of this change to tease out requirements list dependencies.

File: requirements/common_constraints.txt (hunk 1 above)
Comment (schenedx) [reply requirements/common_constraints.txt]: Removed the duplicates from the PR. I'll investigate why `make upgrade` dupes this. That should be outside the scope of this PR tho.

**Review by alangsto — APPROVED** [review]
👍

**Pull request merged**